### PR TITLE
Display worker execution error in A4C log debug

### LINKED ISSUE #503: A4C Logs are displaying stack error when workflow step fails
# Bug Report
Slurm job Logs are displaying stack error
## Description
FastMLe context , I am launching a training and ask for a gpu and it is impossible as I am on  VM on Openstack. 
Looking at the alien4cloud logs (from the GUI) ,  I can see an error as expected : "sbatch: error: Batch job submission failed: Requested node configuration is not available" 
but there are also the full stack error (see below) that doesn't help the user to understand the root cause of the problem

We would like to see the error and not the full stack

## Actual behavior

```

Here the alien4cloud logs 

 [2019-08-20 16:35:29] [run] [] Status for workflow "run" changed to "initial"
  [2019-08-20 16:35:29] [run] [] Status for workflow "run" changed to "running"
  [2019-08-20 16:35:29] [run] [] Start processing workflow step run:TrainingJobComp_submitting
  [2019-08-20 16:35:29] [run] [TrainingJobComp] Status for node "TrainingJobComp", instance "0" changed to "submitting"
  [2019-08-20 16:35:29] [run] [] DeploymentID:"FMLE__c0e12d67_4c1f_4ceb_a25f_0d873bcd326c-Environment", Workflow:"run", step:"TrainingJobComp_submitting" ended without error
  [2019-08-20 16:35:29] [run] [] Start processing workflow step run:TrainingJobComp_submit
  [2019-08-20 16:35:29] [run] [TrainingJobComp] Status for workflow step "TrainingJobComp_submit" changed to "INITIAL"
  [2019-08-20 16:35:29] [run] [TrainingJobComp] Status for task execution "5df269e0-107f-4690-8753-a7b5bfea87c8-0" changed to "INITIAL"
  [2019-08-20 16:35:29] [run] [TrainingJobComp] Status for workflow step "TrainingJobComp_submit" changed to "RUNNING"
  [2019-08-20 16:35:29] [run] [TrainingJobComp] Status for task execution "5df269e0-107f-4690-8753-a7b5bfea87c8-0" changed to "RUNNING"
  [2019-08-20 16:35:31] [run] [TrainingJobComp] sbatch: error: Memory specification can not be satisfied
**sbatch: error: Batch job submission failed: Requested node configuration is not available
: Process exited with status 1**
  [2019-08-20 16:35:31] [run] [TrainingJobComp] Status for workflow step "TrainingJobComp_submit" changed to "ERROR"
  [2019-08-20 16:35:31] [run] [TrainingJobComp] Status for task execution "5df269e0-107f-4690-8753-a7b5bfea87c8-0" changed to "ERROR"
  [2019-08-20 16:35:31] [run] [TrainingJobComp] Status for node "TrainingJobComp", instance "0" changed to "error"
  [2019-08-20 16:35:31] [run] [] Error 'Process exited with status 1
github.com/ystia/yorc/v3/helper/sshutil.(*SSHClient).RunCommand
/home/travis/gopath/src/github.com/ystia/yorc/helper/sshutil/sshutil.go:131
github.com/ystia/yorc/v3/prov/slurm.(*executionCommon).submitJob
/home/travis/gopath/src/github.com/ystia/yorc/prov/slurm/execution.go:489
github.com/ystia/yorc/v3/prov/slurm.(*executionCommon).prepareAndSubmitJob
/home/travis/gopath/src/github.com/ystia/yorc/prov/slurm/execution.go:443
github.com/ystia/yorc/v3/prov/slurm.(*executionCommon).execute
/home/travis/gopath/src/github.com/ystia/yorc/prov/slurm/execution.go:179
github.com/ystia/yorc/v3/prov/slurm.(*defaultExecutor).ExecOperation
/home/travis/gopath/src/github.com/ystia/yorc/prov/slurm/executor.go:81
github.com/ystia/yorc/v3/tasks/workflow.(*step).runActivity.func3
/home/travis/gopath/src/github.com/ystia/yorc/tasks/workflow/step.go:343
github.com/ystia/yorc/v3/tasks/workflow.(*step).runActivity
/home/travis/gopath/src/github.com/ystia/yorc/tasks/workflow/step.go:344
github.com/ystia/yorc/v3/tasks/workflow.(*step).run.func3
/home/travis/gopath/src/github.com/ystia/yorc/tasks/workflow/step.go:210
github.com/ystia/yorc/v3/tasks/workflow.(*step).run
/home/travis/gopath/src/github.com/ystia/yorc/tasks/workflow/step.go:227
```

## Steps to reproduce the issue

1. Step1
launch a job with resource (gpu f.e) not available on the platform
2. Step2 ...
look at the alien4cloud logs

## Yorc configuration file
yorc --version
yorc version v3.2.2

### PR DESCRIPTION
**Pull request by stebenoist**
Remove duplicated error A4C log

# Pull Request description

### What I did
- Remove duplicated error log in runWorkflowStep function
- Change log level to DEBUG for displaying error stack trace.

### How to verify it
See #503 
### Description for the changelog
* Slurm job Logs are displaying stack error ([GH-460](https://github.com/ystia/yorc/issues/503))
## Applicable Issues
Fixes #503

### CODE REVIEW

File: tasks/workflow/worker.go [hunk 1]
```diff
@@ -227,7 +227,7 @@ func (w *worker) handleExecution(t *taskExecution) {
 		err = errors.Errorf("Unknown TaskType %d (%s) for TaskExecution with id %q", t.taskType, t.taskType.String(), t.taskID)
 	}
 	if err != nil {
-		events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelERROR, t.targetID).Registerf("%v", err)
+		events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelDEBUG, t.targetID).Registerf("%v", err)
```
Comment (laurentganne) [inline tasks/workflow/worker.go]: Couldn't we keep the log level to ERROR here, and have instead a change in `helper/sshutil/sshutil.go` RunCommand() function
to have RunCommand() log at debug level the value of errors.WithStack(err) but have this function RunCommand() return directly err.
As err comes from an error running a command, we could just return this error in RunCommand() without the stack trace

Comment (stebenoist) [reply tasks/workflow/worker.go]: Yes, I thought first it was the solution but I didn't see any change when I tested it except we loose first stack trace steps.

Comment (stebenoist) [reply tasks/workflow/worker.go]: Maybe it can be useful to keep it anyway

Comment (loicalbertin) [reply tasks/workflow/worker.go]: What will be the resulting logs if we have something like this?:
```suggestion
		events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelERROR, t.targetID).Registerf("%v", err)
		events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelDEBUG, t.targetID).Registerf("%+v", err)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## UNRELEASED
 
+### BUG FIXES
+
+* Slurm job Logs are displaying stack error ([GH-460](https://github.com/ystia/yorc/issues/503))
```
Comment (loicalbertin) [inline CHANGELOG.md]: This is not specific to Slurm jobs....

File: tasks/workflow/worker.go (hunk 1 above)
Comment (stebenoist) [reply tasks/workflow/worker.go]: Same logs with "%v" and "%+v"

Comment (loicalbertin) [reply tasks/workflow/worker.go]: But why???? :scream:

Comment (stebenoist) [reply tasks/workflow/worker.go]: found the problem here: https://github.com/ystia/yorc/blob/bugfix/GH-503-display-stacktrace-in-debug/tasks/workflow/worker.go#L744

File: CHANGELOG.md (hunk 2 above)
Comment (stebenoist) [reply CHANGELOG.md]: Ok I rename the issue

File: tasks/workflow/worker.go [hunk 3]
```diff
@@ -228,9 +228,6 @@ func (w *worker) handleExecution(t *taskExecution) {
 	}
 	if err != nil {
 		events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelERROR, t.targetID).Registerf("%v", err)
```
Comment (loicalbertin) [inline tasks/workflow/worker.go]: I think we could still have details in debug level WDYT?

```suggestion
		events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelERROR, t.targetID).Registerf("%v", err)
		events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelDEBUG, t.targetID).Registerf("%+v", err)
```

Comment (stebenoist) [reply tasks/workflow/worker.go]: Ok

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 language: go
 go:
-  - stable
```
Comment (loicalbertin) [inline .travis.yml]: Should go back to stable

File: tasks/workflow/worker.go (hunk 3 above)
Comment (stebenoist) [reply tasks/workflow/worker.go]: Done !

**Pull request merged**